### PR TITLE
Fix rotation of billboarded entities with parents

### LINKED ISSUE #1039: Nametags display broken
The display of nametags in click-on-person-to-display-nametag is broken: it doesn't billboard properly to face you. (The display depends on the relative orientation of your and the other person's avatars.)

### PR DESCRIPTION
**Pull request by HifiExperiments**
fix #1039 

entities with billboardMode != NONE now use their local rotation for billboarding instead of their world orientation (which is affected by their parents)

Test plan:
- billboardMode should behave as it did before and nametags should work

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-PR1041-bd41378.tar_95458f3c.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-PR1041-bd41378.tar_95458f3c.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-PR1041-bd41378_95458f3c.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-PR1041-bd41378_95458f3c.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-PR1041-bd41378_95458f3c.exe

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-Quest-Alpha-PR1041-bd41378_95458f3c.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-PR1041-bd41378_95458f3c.apk

**Comment by Aitolda** [comment]
Billboarding and Nametags now behaving as expected.

**Comment by ctrlaltdavid** [comment]
Rather than completely removing the rotation feature, in EntityTree.cpp, can the application of the rotation be dependent on whether or not the entity is parented?

**Comment by HifiExperiments** [comment]
ok...trying this more complicated fix that should preserve the feature!

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-PR1041-30bbb51_852edec9.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-PR1041-30bbb51.tar_852edec9.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-PR1041-30bbb51_852edec9.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-PR1041-30bbb51.tar_852edec9.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-PR1041-30bbb51_852edec9.exe

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-Quest-Alpha-PR1041-30bbb51_852edec9.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1041/Vircadia-PR1041-30bbb51_852edec9.apk

**Review by digisomni — APPROVED** [review]
So it appears to work as expected now. :)

File: libraries/entities-renderer/src/RenderableEntityItem.h [hunk 1]
```diff
@@ -101,24 +101,22 @@ class EntityRenderer : public QObject, public std::enable_shared_from_this<Entit
...
     inline bool isValidRenderItem() const { return _renderItemID != Item::INVALID_ITEM_ID; }
 
     virtual void setIsVisibleInSecondaryCamera(bool value) { _isVisibleInSecondaryCamera = value; }
     virtual void setRenderLayer(RenderLayer value) { _renderLayer = value; }
     virtual void setCullWithParent(bool value) { _cullWithParent = value; }
 
-signals:
-    void requestRenderUpdate();
-
-protected:
```
Comment (ctrlaltdavid) [inline libraries/entities-renderer/src/RenderableEntityItem.h:115]: Did you really intend to remove "protected"?

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

File: libraries/entities-renderer/src/RenderableEntityItem.h (hunk 1 above)
Comment (HifiExperiments) [reply libraries/entities-renderer/src/RenderableEntityItem.h:115]: there were two separate protected sections, I combined them by moving the signal to the bottom

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by Aitolda** [comment]
Web panels appear to be flipped by default with billboarding on, but because of this feature it's easy to turn them back around, BUT... it also appears there is still some perspective distortion on web panels if [full billboarding](https://i.gyazo.com/8e707a43e842ab76c42f7ac5fb274aff.jpg) is used.  Looks correct on [Yaw only](https://i.gyazo.com/36f3d739761461acbdbefc950df59849.jpg) though.  In truth I doubt we'll be billboarding many web panels, but worth pointing out.

**Comment by HifiExperiments** [comment]
@Aitolda was its rotation set to 0,0,0?  web surfaces aren’t treated differently than other things, so I’d expect this to work, but with the addition of the ability to manually rotate things, maybe we want to make it so in create, when you set the billboard mode, it automatically changes the rotation to 0

**Comment by Aitolda** [comment]
> @Aitolda was its rotation set to 0,0,0? web surfaces aren’t treated differently than other things, so I’d expect this to work, but with the addition of the ability to manually rotate things, maybe we want to make it so in create, when you set the billboard mode, it automatically changes the rotation to 0

Ok.. I was wrong. It was a web panel someone else had placed, but starting from scratch I'm not having those issues.  Sorry about the confusion.

**Comment by digisomni** [comment]
Ah, so everything's a-okay? :)

**Pull request merged**